DFReader.py

```diff
@@ -574,44 +593,45 @@ def _parse_next(self):
             # we should also cope with other corruption; logs
             # transfered via DataFlash_MAVLink may have blocks of 0s
             # in them, for example
-            print("Failed to parse %s/%s with len %u (remaining %u)" % (fmt.name, fmt.msg_struct, len(body), self.remaining))
+            fmt = "Failed to parse %s/%s with len %u (remaining %u)"
+            print(fmt % (fmt.name, fmt.msg_struct, len(body), self.remaining))
         if elements is None:
             return self._parse_next()
         name = null_term(fmt.name)
         if name == 'FMT':
             # add to formats
             # name, len, format, headings
             self.formats[elements[0]] = DFFormat(elements[0],
-                                                 null_term(elements[2]), elements[1],
-                                                 null_term(elements[3]), null_term(elements[4]))
+                                                 null_term(elements[2]),
+                                                 elements[1],
+                                                 null_term(elements[3]),
+                                                 null_term(elements[4]))
 
         self.offset += fmt.len-3
         self.remaining -= fmt.len-3
         m = DFMessage(fmt, elements, True)
         self._add_msg(m)
 
         self.percent = 100.0 * (self.offset / float(self.data_len))
-        
+
         return m
 
+
 def DFReader_is_text_log(filename):
     '''return True if a file appears to be a valid text log'''
-    f = open(filename)
-    ret = (f.read(8000).find('FMT, ') != -1)
-    f.close()
-    return ret
+    with open(filename) as f:
```

Comment:
The Python specification does not guaranteed your file will be closed()...  Dangling resource plus chance that file writes are not flushed.  The results could be different between CPython, Pypy, Jython, etc.  `with` does the right thing on all implementations.